src/components/CreateProvider.svelte

```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+    import {dotApi, storeCurrentAction, transactionSigningAddress} from "$lib/stores";
+    import type {DotApi} from '$lib/storeTypes';
+    import {defaultDotApi} from "$lib/storeTypes";
+    import type {ApiPromise} from "@polkadot/api";
+    import {isLocalhost} from "$lib/utils";
+    import {submitCreateProvider} from "$lib/connections";
+    import TransactionStatus from "$components/TransactionStatus.svelte";
+    import {isFunction} from "@polkadot/util";
+    import {onMount} from "svelte";
+
+    let newProviderName = '';
+    let localDotApi: DotApi = defaultDotApi;
+    let web3FromSource;
+    let web3Enable;
+    let showTransactionStatus = false;
+    export let txnStatuses: Array<string> = [];
+    export let validAccounts = {}
+    export let signingAddress = '';
+    export let cancelAction;
+
+    onMount(async () => {
+        const extension = await import('@polkadot/extension-dapp');
+        web3FromSource = extension.web3FromSource;
+        web3Enable = extension.web3Enable;
+    });
+
+    let localSigningAddress = '0xabcdefbeef';
+    transactionSigningAddress.subscribe(addr => localSigningAddress = addr)
+    dotApi.subscribe((api: DotApi) => localDotApi = api)
+
+    const doCreateProvider = async (_evt: Event) => {
+        if (newProviderName === '') {
+            alert("please enter a Provider Name");
+            return;
+        }
+        if (!localDotApi) {
+            alert('please reconnect to an endpoint');
+            return;
+        }
+        clearTxnStatuses();
+        let endpointURI: string = localDotApi.selectedEndpoint || '';
+        let signingKeys = validAccounts[signingAddress];
+        showTransactionStatus = true;
+        if (isLocalhost(endpointURI)) {
+            await submitCreateProvider(
+                localDotApi.api as ApiPromise,
+                undefined,
+                endpointURI,
+                signingKeys,
+                newProviderName,
+                addNewTxnStatus);
+        } else {
+            if (isFunction(web3FromSource) && isFunction(web3Enable)) {
+                const extensions = web3Enable('Frequency parachain provider dashboard: Creating provider');
+                if (extensions.length !== 0) {
```

Comment:
I think `web3Enable` returns a promise so `extensions.length` is failing for me.

Reply:
Ok, some code got undone as this was actually working earlier.